File: Linked List/206. Reverse Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null)
            return head;
        ListNode rear = null;
        while (head != null) {
            ListNode next = head.next;
            head.next = rear;
            rear = head;
            head = next;
        }
        return rear;
    }
}

File: Linked List/206. Reverse Linked List.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode *reverseList(struct ListNode *head) {
    if (head == NULL)
        return head;
    struct ListNode *rear = NULL;
    while (head != NULL) {
        struct ListNode *next = head->next;
        head->next = rear;
        rear = head;
        head = next;
    }
    return rear;
}